File: backend/shoppro-infra/src/main/java/com/shoppro/service/AcquisitionService.java
package com.shoppro.service;

import com.shoppro.entity.ChannelCode;
import com.shoppro.entity.FissionTask;
import com.shoppro.entity.WelcomeMessage;
import java.util.List;

/**
 * <p>
 * Acquisition Service Interface
 * </p>
 *
 * @author shoppro
 * @since 2023-12-22
 */
public interface AcquisitionService {

    // Channel Codes
    List<ChannelCode> listChannelCodes(Long userId);

    ChannelCode createChannelCode(ChannelCode channelCode);

    ChannelCode updateChannelCode(Long id, ChannelCode channelCode);

    ChannelCode getChannelCodeByType(Long userId, String channelType);

    void deleteChannelCode(Long id);

    String generateQrCode(String content); // Helper to generate QR image

    // Statistics
    List<com.shoppro.entity.ChannelCodeStat> getChannelStats(Long channelCodeId, java.time.LocalDate startDate,
            java.time.LocalDate endDate);

    java.util.Map<String, Object> getGlobalAcquisitionStats(Long tenantId);

    java.util.List<java.util.Map<String, Object>> getChannelDistribution(Long tenantId);

    java.util.List<java.util.Map<String, Object>> getSalesRanking(Long tenantId);

    void recordScan(Long channelCodeId);

    // Welcome Messages
    List<WelcomeMessage> listWelcomeMessages(Long tenantId);

    WelcomeMessage getWelcomeMessage(Long channelCodeId);

    WelcomeMessage saveWelcomeMessage(WelcomeMessage welcomeMessage);

    void deleteWelcomeMessage(Long id);

    // Fission Tasks
    List<FissionTask> listFissionTasks();

    FissionTask createFissionTask(FissionTask task);

    void toggleFissionTaskStatus(Long id);

    String exportChannelStats();
}

File: shoppro-app/src/services/acquisitionService.ts
import { http } from './http';

export interface ChannelCode {
    id: number;
    userId: number;
    channelType?: string;
    channelName: string;
    codeUrl?: string;
    description?: string;
    scanCount: number;
    followCount: number;
    createdAt?: string;
}

export interface WelcomeMessage {
    id?: number;
    channelCodeId: number;
    msgType: 'text' | 'image' | 'link' | 'miniprogram';
    content: string;
    createdAt?: string;
}

export interface FissionTask {
    id: number;
    name: string;
    type: 'poster' | 'lottery' | 'task_bot';
    config: string;
    status: number;
    createdAt?: string;
}

export const acquisitionService = {
    // Channel Codes
    listChannelCodes: (userId: number) => {
        return http.get<ChannelCode[]>('/acquisition/channels', { params: { userId } });
    },
    createChannelCode: (data: Partial<ChannelCode>) => {
        return http.post<ChannelCode>('/acquisition/channels', data);
    },
    deleteChannelCode: (id: number) => {
        return http.delete(`/acquisition/channels/${id}`);
    },
    getChannelByType: (userId: number, type: string) => {
        return http.get<ChannelCode>('/acquisition/channels/type', { params: { userId, type } });
    },

    // Welcome Messages
    getWelcomeMessage: (channelCodeId: number) => {
        return http.get<WelcomeMessage>(`/acquisition/welcome/${channelCodeId}`);
    },
    saveWelcomeMessage: (data: WelcomeMessage) => {
        return http.post<WelcomeMessage>('/acquisition/welcome', data);
    },

    // Fission Tasks
    listFissionTasks: () => {
        return http.get<FissionTask[]>('/acquisition/fission');
    },
    createFissionTask: (data: Partial<FissionTask>) => {
        return http.post<FissionTask>('/acquisition/fission', data);
    },
    toggleFissionTaskStatus: (id: number) => {
        return http.put(`/acquisition/fission/${id}/status`);
    }
};

export default acquisitionService;
